rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -21,6 +21,7 @@
 #include <functional>
 #include <iostream>
 #include <memory>
+#include <vector>
```

Comment:
nitpick: alphabetical ordering, similarly I didn't comment on all of these cases in the first pass.

Reply:
Done.